delfin/drivers/hds/vsp/rest_handler.py

```diff
@@ -0,0 +1,222 @@
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+import threading
+
+import requests
+import six
+from oslo_log import log as logging
+
+from delfin import cryptor
+from delfin import exception
+from delfin.drivers.hds.vsp import consts
+
+LOG = logging.getLogger(__name__)
+
+
+class RestHandler(object):
+    HDSVSP_SESSION_ID = None
+    STORAGE_DEVICE_ID = None
+    DEVICE_MODEL = None
+    SERIALNUMBER = None
+    HDSVSP_SYSTEM_URL = '/ConfigurationManager/v1/objects/storages'
+    HDSVSP_LOGOUT_URL = '/ConfigurationManager/v1/objects/sessions/'
+    HDSVSP_COMM_URL = '/ConfigurationManager/v1/objects/storages/'
+
+    HDSVSP_AUTH_KEY = 'Authorization'
+
+    def __init__(self, restclient):
+        self.rest_client = restclient
+        self.session_lock = threading.Lock()
+
+    def call(self, url, data=None, method=None):
+        try:
+            res = self.rest_client.do_call(url, data, method,
+                                           calltimeout=consts.SOCKET_TIMEOUT)
+            if res is not None:
+                if (res.status_code == consts.ERROR_SESSION_INVALID_CODE
+                        or res.status_code ==
+                        consts.ERROR_SESSION_IS_BEING_USED_CODE):
+                    LOG.error("Failed to get token=={0}=={1},get token again"
+                              .format(res.status_code, res.text))
+                    # if method is logout,return immediately
+                    if method == 'DELETE' and RestHandler.\
+                            HDSVSP_LOGOUT_URL in url:
+                        return res
+                    self.rest_client.rest_auth_token = None
+                    access_session = self.login()
+                    if access_session is not None:
+                        res = self.rest_client. \
+                            do_call(url, data, method,
+                                    calltimeout=consts.SOCKET_TIMEOUT)
+                    else:
+                        LOG.error('Login res is None')
+                elif res.status_code == 503:
+                    raise exception.InvalidResults(res.text)
+            else:
+                LOG.error('Rest exec failed,the result in none')
+
+            return res
+
+        except Exception as e:
+            err_msg = "Get RestHandler.call failed: %s" % (six.text_type(e))
+            LOG.error(err_msg)
+            raise exception.InvalidResults(err_msg)
+
+    def get_resinfo_call(self, url, data=None, method=None, resName=None):
```

Comment:
If this function is going to 'get' resource info, should the method always be 'GET'?
If the method always be 'GET', we do not need to set method as parameter.

Reply:
not always 'GET'

Reply:
For example?

Reply:
I checked that,get_resinfo_call just is GET,have changed